src/Umbraco.Web/Editors/EntityController.cs

```diff
@@ -1,957 +1,1024 @@
+    /// The API controller used for getting entity objects, basic name, icon, id representation of umbraco objects that are based on CMSNode
+    /// </summary>
+    /// <remarks>
+    /// Some objects such as macros are not based on CMSNode
+    /// </remarks>
+    [EntityControllerConfiguration]
+    [PluginController("UmbracoApi")]
+    public class EntityController : UmbracoAuthorizedJsonController
+    {
+
+        /// <summary>
+        /// Configures this controller with a custom action selector
+        /// </summary>
+        private class EntityControllerConfigurationAttribute : Attribute, IControllerConfiguration
+        {
+            public void Initialize(HttpControllerSettings controllerSettings, HttpControllerDescriptor controllerDescriptor)
+            {
+                controllerSettings.Services.Replace(typeof(IHttpActionSelector), new ParameterSwapControllerActionSelector(
+
+                    //This is a special case, we'll accept a String here so that we can get page members when the special "all-members"
+                    //id is passed in eventually we'll probably want to support GUID + Udi too
+                    new ParameterSwapControllerActionSelector.ParameterSwapInfo("GetPagedChildren", "id", typeof(int), typeof(string)),
+                    new ParameterSwapControllerActionSelector.ParameterSwapInfo("GetPath", "id", typeof(int), typeof(Guid), typeof(Udi)),
+                    new ParameterSwapControllerActionSelector.ParameterSwapInfo("GetById", "id", typeof(int), typeof(Guid), typeof(Udi)),
+                    new ParameterSwapControllerActionSelector.ParameterSwapInfo("GetByIds", "ids", typeof(int[]), typeof(Guid[]), typeof(Udi[]))));
+            }
+        }
+
+        private readonly UmbracoTreeSearcher _treeSearcher = new UmbracoTreeSearcher();
+
+        /// <summary>
+        /// Returns an Umbraco alias given a string
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="camelCase"></param>
+        /// <returns></returns>
+        public dynamic GetSafeAlias(string value, bool camelCase = true)
+        {
+            var returnValue = (string.IsNullOrWhiteSpace(value)) ? string.Empty : value.ToSafeAlias(camelCase);
+            dynamic returnObj = new System.Dynamic.ExpandoObject();
+            returnObj.alias = returnValue;
+            returnObj.original = value;
+            returnObj.camelCase = camelCase;
+
+            return returnObj;
+        }
+
+        /// <summary>
+        /// Searches for results based on the entity type
+        /// </summary>
+        /// <param name="query"></param>
+        /// <param name="type"></param>
+        /// <param name="searchFrom">
+        /// A starting point for the search, generally a node id, but for members this is a member type alias
+        /// </param>
+        /// <returns></returns>
+        [Obsolete("This method is obsolete, use the overload with dataTypeId instead", false)]
```

Comment:
You could still make this backwards compatible by just adding an optional `Guid? dataTypeId = null` parameter as an additional param